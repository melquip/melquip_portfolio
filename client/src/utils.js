export const loadState = (keys) => {
	try {
		let serializedState = null;
		let err = false;
		serializedState = keys.reduce((serialSt, key) => {
			serialSt[key] = localStorage.getItem('melquip_' + key);
			if(serialSt[key] !== null) {
				serialSt[key] = JSON.parse(serialSt[key]);
			} else {
				err = true;
			}
			return serialSt;
		}, {})
		return serializedState === null || err ? undefined : serializedState;
	} catch (err) {
		console.error('loadState:', err);
		return undefined;
	}
}

export const saveState = (state, keys) => {
	try {
		let serializedState = null;
		if (keys.length > 0) {
			keys.forEach(key => {
				serializedState = JSON.stringify(state[key]);
				localStorage.setItem('melquip_' + key, serializedState);
			});
		} else {
			serializedState = JSON.stringify(state);
			localStorage.setItem('melquip_state', serializedState);
		}
	} catch (err) {
		throw new Error('Not able to set local storage: ', err);
	}
}