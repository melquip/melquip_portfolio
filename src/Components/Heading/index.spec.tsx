// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { render } from "@Test/utils";
import { Heading } from "@Components";
// #endregion Local Imports

describe("Components", () => {
    describe("Heading", () => {
        it("should render given text", () => {
            const { getAllByText } = render(<Heading text="Render Me!" />);

            expect(getAllByText("Render Me!").length).toEqual(2);
        });

        it("should match snapshot", () => {
            const { container } = render(<Heading text="Test" />);

            expect(container).toMatchSnapshot();
        });
    });
});
