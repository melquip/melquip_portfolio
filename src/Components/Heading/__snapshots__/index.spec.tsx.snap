// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Heading should match snapshot 1`] = `
<div>
  <div>
    <span>
      Test
    </span>
    <span>
      Test
    </span>
  </div>
</div>
`;
